tests/test_applets/structuredLearningTracking/testOpStructuredTracking.py

```diff
@@ -0,0 +1,204 @@
+import pytest
+from hytra.core.hypothesesgraph import HypothesesGraph
+from hytra.core.probabilitygenerator import Traxel
+from ilastik.applets.tracking.structured.opStructuredTracking import (
+    AnnotationHypothesisgraphMismatchException,
+    OpStructuredTracking,
+)
+
+
+@pytest.fixture
+def tracklet_graph():
+    #               /  2    4
+    #      0  ->  1       /
+    #               \  3
+    #                     \
+    #                       5
+    #
+    #             6 -> 7 -> 8
+    #
+    # t:   0      1    2    3
+
+    h = HypothesesGraph()
+    h._graph.add_path([(0, 0), (1, 1), (2, 2)])
+    h._graph.add_path([(1, 1), (2, 3), (3, 4)])
+    h._graph.add_path([(2, 3), (3, 5)])
+    h._graph.add_path([(1, 6), (2, 7), (3, 8)])
+    for n in h._graph.nodes:
+        h._graph.nodes[n]["id"] = n[1]
+        h._graph.nodes[n]["traxel"] = Traxel()
+        h._graph.nodes[n]["traxel"].Id = n[1]
+        h._graph.nodes[n]["traxel"].Timestep = n[0]
+
+    return h
+
+
+@pytest.fixture
+def annotations():
+    """
+    Annotations resulting in the following graph:
+
+                   /  2    4
+          0  ->  1        /
+                   \  3 =
+                          \
+                           5
+
+     t:   0      1    2    3
+
+    = denote divisions
+
+    annotations format:
+        labels: {timeframe: {object_id: set(track_ids)}}
+        divisions: {parent_track_id: ([child_track_id1, child_track_id2], timeframe)}
+    """
+    annotations = {
+        "labels": {
+            0: {0: {1, 2}},
+            1: {1: {1, 2}},
+            2: {2: {1}, 3: {2}},
+            3: {4: {3}, 5: {4}},
+        },
+        "divisions": {2: ([3, 4], 2)},
+    }
+    return annotations
+
+
+class InstantTraxel(Traxel):
+    """Convenience, instantiate Traxel with timestep and id, add __eq__"""
+
+    def __init__(self, timestep, uid):
+        super().__init__()
+        self.Timestep = timestep
+        self.Id = uid
+
+    def __eq__(self, other):
+        if not isinstance(other, Traxel):
+            return False
+        if self.Timestep == other.Timestep:
+            if self.Id == other.Id:
+                return True
+
+        return False
+
+
+@pytest.fixture
+def expected():
```

Comment:
Do we need to keep "nodes" and "edges" structures together? We could split them into 2 different fixtures.

Reply:
you're absolutely right. I kept them together as I was just understanding the tracking code, and there annotations are kept together in a dict. But here there is no benefit, other than mirroring the data structures used in the code.

Reply:
Oh, I missed that, sorry. If it mirrors a structure in SUT, then definitely keep it as-is!